deps/url.lua

```diff
@@ -25,22 +25,28 @@ function exports.parse(url, parseQueryString)
   local chunk, protocol = url:match("^(([a-z0-9+]+)://)")
   url = url:sub((chunk and #chunk or 0) + 1)
 
-  local chunk, auth = url:match('(([0-9a-zA-Z]+:?[0-9a-zA-Z]+)@)') 
+  chunk, auth = url:match('(([0-9a-zA-Z]+:?[0-9a-zA-Z]+)@)') 
```

Comment:
`auth` here is a leaking global.  I highly recommend running luacheck on all lua code to prevent scope leakage.  I'm using sublimelinter with luacheck in my workflow and it's super helpful.